sor/src/main/java/com/bazaarvoice/emodb/sor/db/DeltaIterator.java

```diff
@@ -0,0 +1,201 @@
+package com.bazaarvoice.emodb.sor.db;
+
+
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.Lists;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.UUID;
+
+
+/**
+ *  Iterator abstraction that stitches blocked deltas together
+ */
+abstract public class DeltaIterator<R, T> extends AbstractIterator<T> {
+
+    private List<R> _list;
+
+    private final Iterator<R> _iterator;
+    private R _next;
+    private R _oldDelta;
+    private final boolean _reverse;
+    private final int _prefixLength;
+
+    public DeltaIterator(Iterator<R> iterator, boolean reverse, int prefixLength) {
+        _iterator = iterator;
+        if (iterator.hasNext()) {
```

Comment:
Don't call `iterator.hasNext()` and `iterator.next()` in the constructor.  This is meant to wrap what is potentially an asynchronously-populated or never-iterated iterator, and these calls force a synchronous wait for the first item upon construction.  Refactor so the initialization takes place on the first call to `computeNext()`.

Reply:
done in migration

Reply:
done